plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/AbstractMetastoreTableOperations.java

```diff
@@ -105,6 +106,7 @@ protected final void commitNewTable(TableMetadata metadata)
                 .withStorage(storage -> storage.setStorageFormat(ICEBERG_METASTORE_STORAGE_FORMAT))
                 // This is a must-have property for the EXTERNAL_TABLE table type
                 .setParameter("EXTERNAL", "TRUE")
+                .setParameter(PRESTO_QUERY_ID_NAME, session.getQueryId())
```

Comment:
I'm not sure whether this is necessary.

Take a look whether `metadata.currentSnapshot().summary()` contains already `trino_query_id` .

In this way we could apply the fix potentially for all the implementations instead of doing it only for Hive metastore.